src/components/structures/MatrixChat.js

```diff
@@ -635,23 +657,35 @@ module.exports = React.createClass({
                 action: 'start_post_registration',
             });
         } else if (screen.indexOf('room/') == 0) {
-            var roomString = screen.split('/')[1];
+            var roomString = screen.substring(5);
+            var eventId;
+
+            // extract event id, if one is given
+            var idx = roomString.indexOf('/');
+            if (idx >= 0) {
+                eventId = roomString.substring(idx+1);
+                roomString = roomString.substring(0, idx);
```

Comment:
Yes, the idea is to support `/room/{roomId}` and `/room/{roomId}/{eventId}`.  I'm not quite sure what you mean, though. What should I do if there are more `/` characters? Why is assuming that we can ignore other components better?